### ISSUE TITLE
Somehow use short cut keys for operations? Or expand the keyboard macros for the master keyboard?

### ISSUE THREAD

**Issue by christofmuc** [christofmuc/KnobKraft-orm#129]
We currently have the keyboard macros on the masterkeyboard screen where you can assign hide to a chord or key, but of course we could also allow function keys/ctrl keys... how to set this up and teach the novice user?

**Comment by markusschloesser** [comment]
my suggestion: use OS level (and therefor known) keyboard shortcuts. imho no need to map or do macro. JUCE probably has something for this

**Comment by markusschloesser** [comment]
possible targets:
1. del = delete
2. h = hide
3. f = favorite
4. the usual ctrl/cmd - c/x/v
5. F2 / CTRL - R = Rename
6. CTRL - O = open database
7. CTRL - N = new database
8. CTRL - +/- = scale view
9. CTRL - I = import patch from synth
10. CTRL - E = import edit buffer
11. CTRL - 1/2/3 etc = switch to first/second/third synth

**Comment by bboc** [comment]
Ctrl/Cmd+F could be find, that shortcut is  used.
Maybe 1-9 could be used for selecting a bank of the current synth
if Ctrl/Cmd+1-9 is used for switching synths, the button for each synth might show that number.

**Comment by planist79** [comment]
Hi,
instead of opening up a new issue i try my luck here:

Is it possible to assign keyboard shortcuts to "next patch / previous patch" (for example to arrow left/right) in Synthbank view or Database view - instead of clicking patches with the mouse? This would make changes patches feel like changing it on the front panel of the synth.

**Comment by christofmuc** [comment]
Yes, this is a sensible request and should probably be done together with the other suggestions here. Shouldn't be too hard, but my experiences with JUCE and hotkeys are a bit mixed, as the focus handling sometimes is quirky and your key presses go into the Nirwana.

**Comment by RadekPilich** [comment]
> We currently have the keyboard macros on the masterkeyboard screen where you can assign hide to a chord or key, but of course we could also allow function keys/ctrl keys... how to set this up and teach the novice user?

I would suggest first to add a Shorcuts / Chord-App Shorcuts or some descriptive heading into the Macros section above the shortcuts. 

In my case, it's partially the absence of such heading that lead to me not being curious enough about what that section is about.

**Comment by christofmuc** [comment]
The lowest/highest are just so you can define the range of the keyboard in front of you, so you don't define macros that are not availabel on the 4 octave synth or so... for the old synths (C++ code) those values were populated automatically, so it would show me the correct 5 octaves for the Rev2 and 4 octaves for the OB-6. This has been forgotten a bit, and would need support for each adaptation... 

I am not sure I understood your suggestion though - is this about the documentation on the macros screen?

**Comment by RadekPilich** [comment]
Anhhh, so the notes range is randomly correct for Summit as well, so I was expecting some additional function.

Yes, the suggestion was about macros screen, about GUI rather then documentation. Just add one additional collapsible header section for the shortcuts mapping part of the screen. To make it more obvious what's that about.
